src/models/ArticleSummaryRequest.js

```diff
@@ -0,0 +1,27 @@
+(function(exports){
+
+  var ArticleSummaryRequest = function(url){
+    this.summary = '';
+    this.articleUrl = url
+  }
+
+  ArticleSummaryRequest.prototype.requestSummary = function(callback,HttpRequest = new XMLHttpRequest){
+    var xhttp = HttpRequest
+    var self = this
+    xhttp.onreadystatechange = function(){
+      if (xhttp.readyState == 4 && xhttp.status === 200) {
+          var data = JSON.parse(xhttp.responseText)
+          self.summary = data.sentences.join()
```

Comment:
It feels like this is a separate responsibility. We have two things going on here - sending the request and acting on the response. I would suggest that you just pass the data to the callback, e.g:

```
var data = JSON.parse(xhttp.responseText);
callback(data);
```

It's the responsibility of the callback to know what to do with that data. This is a pretty standard pattern for API calls in javascript.